leshan-core/src/main/java/org/eclipse/leshan/core/request/UpdateRequest.java

```diff
@@ -52,6 +57,10 @@ public UpdateRequest(String registrationId, Long lifetime, String smsNumber, Bin
         this.lifeTimeInSec = lifetime;
         this.bindingMode = binding;
         this.smsNumber = smsNumber;
+        if (additionalAttributes == null)
+            this.additionalAttributes = Collections.emptyMap();
+        else
+            this.additionalAttributes = Collections.synchronizedMap(new HashMap<>(additionalAttributes));
```

Comment:
`UpdateRequest` should be immutable.

Reply:
I don't get it... which kind of Map should I use?

Reply:
Collections.unmodifiableMap(new HashMap<>(additionalAttributes));